src/app/OrderButton.tsx

```diff
@@ -95,21 +95,16 @@ export function OrderButton() {
       account &&
       gatewayApi
     ) {
+      // TODO: Move all logic to separate pure TS class/classes/file/files.
+      // Context providers will recieve only the data necessary to render the page.
+      // This will allow cleaner code and better incapsulation.
```

Comment:
Perhaps we could do the trick with redux toolkit (it's not that horrible as the good old vanilla redux) 🤔

Reply:
The idea is to prevent errors like that from happening again.

Reply:
@fliebenberg @SmashingBumpkin @maelsar are you OK if I refactor the webpage to use https://redux-toolkit.js.org/?

Yes, I was previously opposed to using redux, but since we have so many interconnected state variables I see the point.

# The rationale for refactoring
I just discovered another [similar bug ](https://github.com/DeXter-on-Radix/website/issues/69) having the same origins: Component implementation is mixed with business logic, and so the  business logic contains errors that are hard to investigate/debug/unit-test.

Reply:
Alternatively, we could use [useReducer hook](https://react.dev/learn/extracting-state-logic-into-a-reducer) 🤔  I'll experiment with that and will try to fix all the state-related bugs in this PR, converting to draft.

Reply:
I've experimented a little bit with the built-in useReducer and while it is plausible, it adds almost as much boilerplate as redux-toolkit. Other reasons I now prefer react-toolkit for the website:
- separating API calls from the view will make the code more readable and we will be able to write unit tests
- single state for all app (it appears many components reuse the same updates)
- it will be harder for us to create state-related bugs (redux uses immer.js)
- debugging is easier with the [chrome extension](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd)

My hope is to make our lives easier in the long run, apologies for possible code conflicts. Here is the branch I'm doing it in [refactor-redux-toolkit](https://github.com/DeXter-on-Radix/website/tree/refactor-redux-toolkit) - all the components will have dedicated slices that will be responsible for data fetching and converting to values for the views, and the views themselves will just render whatever values are in the state. 

Let me know what you think, if you don't want to work with redux in the future, I can stop and go back to fixing OrderButton state as it is. @SmashingBumpkin @maelsar @fliebenberg